server/common/common.go

```diff
@@ -0,0 +1,45 @@
+package common
```

Comment:
I don't like "common" as a package. What is it common to? What advantage do these very generic types give us? (Also this looks less "common" and more "json")

Reply:
I think what I would want here if we're keeping the "common" package is a *very clear* package description that explicitly says what it is for.